src/main/java/org/testng/IDataProviderMethod.java

```diff
@@ -34,4 +34,9 @@
      */
     List<Integer> getIndices();
 
+    /**
+     * @return - The {@link IFactoryMethod} to which the data was injected for instantiating an object.
+     */
+    IFactoryMethod getFactoryMethod();
```

Comment:
A factory method may have or not a data provider, but a data provider may have no or many constructors and/or methods

Reply:
I didn't quite understand this comment. Can you please let me know as to what you are hinting at here ? Are you suggesting a verbiage change of the javadocs ?

Reply:
In fact, the modelization looks wrong: I understand `IFactoryMethod#getDataProvider()` but the same data provider may be used by different factories.

Reply:
Yes I agree. Its a one to one relationship between a factory method and a data provider [ we aren't going to be considering situations wherein a factory method doesn't involve a data provider, because in that situation the new `IDataProviderListener` is not involved at all], which is what I am trying to capture.

But when it comes to a data provider to factory, it can be either a one to one or a one to many. 

So I don't quite understand how the modelization looks wrong.

Reply:
I think it is a one to many relation. So it should be: `List<IFactoryMethod> getFactories()` in `IDataProvider` or `IDataProviderMethod getDataProvider()` in `IFactoryMethod`. I prefer the 2nd option.

Reply:
@juherr - I think i would have to close this PR for now, because I am not clear on what the expectations are for this issue. It would be good if you could please help add what all you think should be captured as part of issue #1515. That way it would be easy for me to start work on this.